src/objects/domain_obj.f90

```diff
@@ -1016,20 +1016,26 @@ subroutine setup_sleve(this, options)
             ! ( Although an argument could be made to calculate this on the offset (u/v) grid b/c that is most
             !   relevant for advection? In reality this is probably a sufficient approximation, as long as we
             !   aren't pushing the gamma factor too close to zero )
-            allocate(gamma_n(this%kds : this%kde+1))
+            ! allocate(gamma_n(this%kds : this%kde+1))
+            allocate(gamma_n(this%kms : max_level))  ! max_level +1 maybe??
+            ! if (options%parameters%debug .and. this_image()==1) write(*,*) " gamma_n: ", gamma_n
+
             i=kms
             gamma_n(i) =  1                                                     &
                 - MAXVAL(h1) * n/(s1**n)                                        &
                 * COSH((smooth_height/s1)**n) / SINH((smooth_height/s1)**n)     &
                 - MAXVAL(h2) * n/(s2**n)                                        &
                 * COSH((smooth_height/s2)**n) / SINH((smooth_height/s2)**n)
+            if (options%parameters%debug .and. this_image()==1) write(*,*) " k, gamma: ", i, gamma_n(i)
 
-            do i = this%grid%kds, this%grid%kde
+            ! do i = this%grid%kds, this%grid%kde 
+            do i = this%grid%kms, max_level-1 ! account for flat z < 0, otherwise gamma can blow up if flat z < 0.
```

Comment:
does gamma_n get used anywhere or is it purely diagnostic?

Reply:
purely diagnostic